pkg/verifier/aws/entry_point.go

```diff
@@ -46,12 +50,49 @@ func (a *AwsVerifier) ValidateEgress(vei verifier.ValidateEgressInput) *output.O
 
 	// Terminate a debug instance leftover from a previous run
 	if vei.TerminateDebugInstance != "" {
+
+		//Terminate the debug instance
 		if err := a.AwsClient.TerminateEC2Instance(vei.Ctx, vei.TerminateDebugInstance); err != nil {
 			a.Output.AddError(err)
 		}
+
+		//Delete the debug keypair
+		_, err := a.AwsClient.DeleteKeyPair(vei.Ctx, &ec2.DeleteKeyPairInput{
```

Comment:
This is called before ImportKeyPair is so this will likely error most of the time. Additionally, this should only be called if vei.ImportKeyPair is specified, otherwise this code will be ran regardless of if a key pair is specified or not

Reply:
Made the changes to the code to check for the key pair existence before executing a deletion.